android/src/main/java/com/oblador/keychain/KeychainModule.java

```diff
@@ -103,11 +109,17 @@ public void getGenericPasswordForOptions(String service, Promise promise) {
                 // decrypt using the older cipher storage
                 decryptionResult = oldCipherStorage.decrypt(service, resultSet.usernameBytes, resultSet.passwordBytes);
                 // encrypt using the current cipher storage
-                EncryptionResult encryptionResult = currentCipherStorage.encrypt(service, decryptionResult.username, decryptionResult.password);
-                // store the encryption result
-                prefsStorage.storeEncryptedEntry(service, encryptionResult);
-                // clean up the old cipher storage
-                oldCipherStorage.removeKey(service);
+
+                try {
```

Comment:
do you think you could refactor the entire if / else part into a separate method (that will return a `decryptionResult` I suppose)? `getGenericPasswordForOptions` is becoming a bit too lengthy. Thanks.

Reply:
done